src/addons/grids/ReactDataGrid.js

```diff
@@ -480,9 +482,27 @@ const ReactDataGrid = React.createClass({
     // we need to prevent default as we control grid scroll
     // otherwise it moves every time you left/right which is janky
     e.preventDefault();
-    let rowIdx = this.state.selected.rowIdx + rowDelta;
+    let rowIdx = this.state.selected.rowIdx;
     let idx = this.state.selected.idx + cellDelta;
-    this.onSelect({idx: idx, rowIdx: rowIdx});
+    let _rowDelta = rowDelta;
+    const { enableCellSelectionLoop } = this.props;
+    if (cellDelta > 0) {
```

Comment:
Think you complicated a little bit in this one don't you? 

1. Why run this piece of code when `enableCellSelectionLoop` is false in the first place? It will just make the if/else chain grow which equals more complexity 
3. For the sake of readability can you please put the if conditions that check state in a new method? For example replacing `this.state.selected.idx === this.props.columns.length - 1 && this.state.selected.rowIdx < this.props.rowsCount - 1` for a method called `isFinalCellInTheRowAndNotLastRow()`?
2. The enable/disable is not working right and we want to keep this turned off by default, replacing that piece of code for something like this will do the trick

```
    if (enableCellSelectionLoop) {
      if (cellDelta > 0 && this.isFinalCellInTheRowAndNotLastRow()) {
        _rowDelta = rowDelta + 1;
        idx = 0;
      } else if (cellDelta < 0 && this.isFirstCellInTheRowAndNotFirstRow()) {
        _rowDelta = rowDelta - 1;
        idx = this.props.columns.length - 1;
      }
    }
```